mlir-sycl/lib/Conversion/SYCLToLLVM/SYCLToLLVMPass.cpp

```diff
@@ -52,9 +55,20 @@ void ConvertSYCLToLLVMPass::runOnOperation() {
 
   RewritePatternSet patterns(context);
 
+  if (useBarePtrCallConv) {
+    // Keep these at the top; these should be run before the rest of
+    // function conversion patterns.
+    populateReturnOpTypeConversionPattern(patterns, converter);
+    populateCallOpTypeConversionPattern(patterns, converter);
+    populateAnyFunctionOpInterfaceTypeConversionPattern(patterns, converter);
+  }
+
   sycl::populateSYCLToLLVMConversionPatterns(converter, patterns);
   populateFuncToLLVMConversionPatterns(converter, patterns);
 
+  if (useBarePtrCallConv)
+    polygeist::populateBareMemRefToLLVMConversionPatterns(converter, patterns);
```

Comment:
These are the relevant changes. Changes to CMake files will have to be adapted after #8357 is merged.